arroyo-compiler-service/src/main.rs

```diff
@@ -268,33 +330,44 @@ impl CompilerGrpc for CompileService {
 
     async fn check_udfs(
         &self,
-        request: Request<CheckUdfsReq>,
+        request: Request<CheckUdfsCompilerReq>,
     ) -> Result<Response<CheckUdfsCompilerResp>, Status> {
         // only allow one request to be active at a given time
         let _guard = self.lock.lock().await;
 
+        let req = request.into_inner();
+
         info!("Checking UDFs");
         let start = Instant::now();
 
-        // write udf to build_dir udfs module
-        let build_dir = &self.build_dir;
-        tokio::fs::write(
-            build_dir.join("udfs/src/lib.rs"),
-            &request.into_inner().definition,
-        )
-        .await
-        .unwrap();
+        let Some(udf_crate) = req.udf_crate else {
+            return Err(Status::internal("No UDF crate provided"));
+        };
+
+        let name = udf_crate.name.clone();
+        let udf_build_dir = self.build_dir.join("udfs_dir").join(&name);
+        self.write_udf_crates(&vec![udf_crate])
+            .await
+            .map_err(|e| Status::internal(format!("Writing UDFs failed: {}", e)))?;
 
         let output = Command::new("cargo")
-            .current_dir(&self.build_dir)
+            .current_dir(&udf_build_dir)
             .arg("check")
-            .arg("--package=udfs")
+            .arg(format!("--package={}", &name))
             .arg("--message-format=json")
             .output()
             .await?;
 
+        // clean up the udfs_dir
+        if !self.debug {
```

Comment:
why do we have this check?

Reply:
Looking at the generated UDF crates was helpful during development, so I figured I'd make that possible in debug mode.

Reply:
I've changed it to always clean up the directory, and to do it before checking/compiling.